### PR TITLE
add ability for users to leave reviews, angularize UI, display reviews on profile

### PR DESCRIPTION
**Pull request by danyag0820**

### CODE REVIEW

File: portfolio/src/main/webapp/book-detail/book-detail.template.html [hunk 1]
```diff
@@ -25,6 +25,19 @@ <h2>{{bookDetailCtrl.book.title}}</h2>
                 <a ng-href="#!/book-detail/{{eachBook.isbn}}">{{eachBook.title}}</a> 
             </li>
         </p>
+
+        <p class="booklst">Reviews: </p>
```
Comment (saur-g) [inline portfolio/src/main/webapp/book-detail/book-detail.template.html:29]: Is it possible to hide this form when the user has already left a review for this book? You can do this in separate PR.

File: portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java [hunk 2]
```diff
@@ -51,4 +52,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    
+    String userReviewText = request.getParameter("reviewText");
+    String isbn = request.getParameter("isbn");
+    String email = userService.getCurrentUser().getEmail();
+    
+    Review userReview = Review.create(userReviewText,isbn,email);
+    try {
+        reviewDao.uploadNew(userReview);
+    } catch (Exception ex) {
+        System.out.println(ex + ", Review for this book by the user already exists");
```
[the remark this answers is not in the document]
Comment (danyag0820) [reply portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java]: sounds good!

File: portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java [hunk 3]
```diff
@@ -51,4 +53,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    String userReviewText = request.getParameter("reviewText");
+    String isbn = request.getParameter("isbn");
+    String email = userService.getCurrentUser().getEmail();
+    
+    Review userReview = Review.create(userReviewText,isbn,email);
```
Comment (cjwert-g) [inline portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java:65]: nit: spaces after the commas between params

File: portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java [hunk 4]
```diff
@@ -51,4 +53,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    try {
+        reviewDao.uploadNew(userReview);
+    } catch (ServletException ex) {
```
Comment (cjwert-g) [inline portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java:68]: nit: this middle catch block for ServletException might be a little redundant since it does the exact same as the Exception catch block. 

ServletException class inherits from Exception (https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/ServletException.html), so I would think the 'Exception' block would still catch ServletExceptions.

File: portfolio/src/main/webapp/about-us/about-us.module.js [hunk 5]
```diff
@@ -0,0 +1 @@
+angular.module('aboutUs', []);
```
Comment (cjwert-g) [inline portfolio/src/main/webapp/about-us/about-us.module.js:1]: nit: EOF newline

Comment (cjwert-g) [reply portfolio/src/main/webapp/about-us/about-us.module.js:1]: Also missing in some other files in this PR.

File: portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java (hunk 4 above)
Comment (saur-g) [reply portfolio/src/main/java/com/google/sps/servlets/UserReviewServlet.java]: I might have confused you, I think we want to throw the servlet exception here.
``` 
   try {
      reviewDao.uploadNew(userReview);
    } catch (Exception ex) {
      throw new ServletException("Failed to save review, please ensure review doesn't already exists for this book", ex);
    }
```
